### ISSUE TITLE
support for vanity imports

### ISSUE THREAD

**Issue by mkmik** [brettlangdon/git-vendor#8]
e.g. https://github.com/sourcegraph/appdash should go in `vendor/sourcegraph.com/sourcegraph/appdash`

perhaps another flag that can force the dir location (name doesn't)

**Comment by brettlangdon** [comment]
@mmikulicic thanks for the suggestion, seems do-able. My Apologies, I might not be able to get to this right away, worst case early next week.

I have two thoughts, we can either add another switch, similar to `--prefix`, which just overrides the directory where it is cloned to, e.g. `git vendor add --dir sourcegraph.com/sourcegraph/appdash` (this will still honor the `--prefix` of `vendor/`). Or maybe `--namespace` or something is a better name for it?

Or.. maybe we can just add as an additional positional argument? `git vendor add <name> <repo> [<version> [<dir>]]` so we could have `git vendor add appdash https://github.com/sourcegraph/appdash master vendor/sourcegraph.com/sourcegraph/appdash`.

Thoughts?

**Comment by mkmik** [comment]
don't know what the best name for the flag would be; I made a local change to the script that just does:

```
    # dir="$prefix/$(echo "$repository" | sed -E 's/^[a-zA-Z]+((:\/\/)|@)//' | sed 's/:/\//' | sed -E 's/\.git$//')"
    dir="$prefix/$name"
```

and it does what need but I guess it's better to not overload the `name` for that as well, given that people might use name for something else

**Comment by brettlangdon** [comment]
yeah, originally I meant for the name just as a human friendly way to display a dependencies information, and to make updates easier, since you can use the short name instead of the full repo path.

ok, but it seems like flag might be best here. I'll think on what the name should be, I feel like I am blanking on the appropriate term for the path/namespace is.

**Comment by mkmik** [comment]
package?

**Comment by brettlangdon** [comment]
yeah, that could work.

**Comment by tv42** [comment]
The right name is import path. https://golang.org/ref/spec#Import_declarations
